genesis/genesis.go

```diff
@@ -0,0 +1,320 @@
+package genesis
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+const decodeBase = 10
+
+// Genesis hold data for decoded data from json file
+type Genesis struct {
+	initialBalances []*InitialBalance
+	entireSupply    *big.Int
+}
+
+// NewGenesis creates a new decoded genesis structure from json config file
+func NewGenesis(
+	genesisFilePath string,
+	entireSupply *big.Int,
+) (*Genesis, error) {
+
+	if entireSupply == nil {
+		return nil, ErrNilEntireSupply
+	}
+	if big.NewInt(0).Cmp(entireSupply) >= 0 {
+		return nil, ErrInvalidEntireSupply
+	}
+
+	initialBalances := make([]*InitialBalance, 0)
+	err := core.LoadJsonFile(&initialBalances, genesisFilePath)
```

Comment:
this will be hard to test :dagger:

Reply:
Agree, added testdata files as to check (also) the happy path behavior as long as some edge cases: file not found or a file containing a bad config. Code coverage is at 100% in this file.